lldb/source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -278,11 +308,13 @@ bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
     if (count > maxCount)
       return false;
 
-    uint64_t buffer[2] = {raw0, raw1};
+    uint64_t rawBuffer[2] = {raw0, raw1};
+    auto *buffer = (uint8_t *)&rawBuffer;
+    applySlice(buffer, count, slice);
```

Comment:
Correct me if I'm wrong but AFAIU this is technically UB. I don't think any compiler will break this kind of type punning but the recommended way of reading a type through a `uint8_t*` is using `memcpy`.

The standard doesn't technically permit doing pointer arithmetic on something that isn't [pointer-interconvertible](http://www.eel.is/c++draft/basic.compound#4). Chances are `memcpy` will get optimised away into exactly the same code that `reinterpret_cast` produces.